### PR TITLE
Use recommended themes for design-picker step

### PR DESCRIPTION
**Pull request by p-jackson**
#### Changes proposed in this Pull Request

* Fetch `auto-loading-homepage` themes using API and display them in design picker step
* Exclude `podcast` themes (the gutenboarding design picker excludes "anchorfm" designs by default)
* Generate a smaller thumbnail if we know the theme isn't intended to have content "below the fold"
* Fetch a higher resolution thumbnail

https://user-images.githubusercontent.com/1500769/134621481-638b5498-21c6-4fd6-b6fe-8c3906af1680.mov

FYI @michaeldcain 

#### Testing instructions

* Start at `/start`, choose free plan, will land on the design picker
* Test with non-`en` user and thumbnails should be translated

### CODE REVIEW

File: client/signup/steps/design-picker/index.jsx [hunk 1]
```diff
@@ -7,8 +7,16 @@ import { connect } from 'react-redux';
...
 import { submitSignupStep } from 'calypso/state/signup/progress/actions';
+import { getRecommendedThemes as fetchRecommendedThemes } from 'calypso/state/themes/actions';
+import { getRecommendedThemes } from 'calypso/state/themes/selectors';
 import './style.scss';
 
+// Ideally this data should come from the themes API, maybe by a tag that's applied to
+// themes? e.g. `link-in-bio` or `no-fold`
+const STATIC_PREVIEWS = [ 'bantry', 'sigler', 'miller', 'pollard', 'paxton', 'jones', 'baker' ];
+
+const SUBJECTS_TO_EXCLUDE = [ 'podcast' ];
```
Comment (michaeldcain) [inline client/signup/steps/design-picker/index.jsx]: I don't know that this is necessarily the same as an Anchor theme. I believe the ones excluded from the design picker are templates that require Anchor functionality, whereas Spearhead is a (newer and popular) blog-style theme made for podcasters. Let's double-check before excluding that.

File: client/signup/steps/design-picker/index.jsx [hunk 2]
```diff
@@ -7,8 +7,16 @@ import { connect } from 'react-redux';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
 import StepWrapper from 'calypso/signup/step-wrapper';
 import { submitSignupStep } from 'calypso/state/signup/progress/actions';
+import { getRecommendedThemes as fetchRecommendedThemes } from 'calypso/state/themes/actions';
+import { getRecommendedThemes } from 'calypso/state/themes/selectors';
 import './style.scss';
 
+// Ideally this data should come from the themes API, maybe by a tag that's applied to
+// themes? e.g. `link-in-bio` or `no-fold`
+const STATIC_PREVIEWS = [ 'bantry', 'sigler', 'miller', 'pollard', 'paxton', 'jones', 'baker' ];
```
Comment (michaeldcain) [inline client/signup/steps/design-picker/index.jsx:16]: Does this equate to whether or not there's a pattern toolkit site for the theme?

File: client/signup/steps/design-picker/index.jsx [hunk 3]
```diff
@@ -26,6 +36,14 @@ class DesignPickerStep extends Component {
 		showOnlyThemes: false,
 	};
 
+	componentDidMount() {
+		this.fetchThemes();
```
Comment (michaeldcain) [inline client/signup/steps/design-picker/index.jsx]: Should we check if `this.props.showOnlyThemes === true`?

**Review by michaeldcain — COMMENTED** [review]
This is looking great and works well! I know it's still a draft, but I left a few comments. Let me know if I can help at all.

File: client/signup/steps/design-picker/index.jsx (hunk 3 above)
Comment (p-jackson) [reply client/signup/steps/design-picker/index.jsx]: 🆗 yeah we should. I think we can get rid of `showOnlyThemes` actually, it's only there so we don't break another flow, but that flow is no longer needed and can be removed. That's another PR though.

File: client/signup/steps/design-picker/index.jsx (hunk 2 above)
Comment (p-jackson) [reply client/signup/steps/design-picker/index.jsx:16]: These are the themes that shouldn't have the hover effect e.g. the intention for Bantry is that everything appears above the fold. The design picker calls this kind of preview "static"
Adding a tag for this sort of theme is discussed here: p1632436419100900-slack-C029SB8JT8S

File: client/signup/steps/design-picker/index.jsx (hunk 1 above)
Comment (p-jackson) [reply client/signup/steps/design-picker/index.jsx]: pdgK6S-be-p2

Comment (p-jackson) [reply client/signup/steps/design-picker/index.jsx]: No special filtering, I'll show all recommended themes: pdgK6S-be-p2#comment-341

**Comment by autumnfjeld** [comment]
Repeating slack chat with @p-jackson 

**Testing**
* Went through /start as an existing user.
* I picked the Ryu theme.
* Theme selection successful
* But when I went through site checklist -> Update Your Homepage I see blue trees! 😱 
* Expected to see Ryu content

Phil response:  
> Oh no, the Ryu theme doesn’t come with a built in annotation!
> I’m going to have to go through and audit the themes to see which don’t have a built in annotation.

**Comment by p-jackson** [comment]
Turns out Ryu is the only recommended theme without annotations. I will explicitly exclude it for now, but will need to P2 about whether this is just an oversight.

**Comment by jordesign** [comment]
Getting stuck on the 'Applying Design' step when selecting Spearhead:
https://container-modest-carson.calypso.live/start/setup-site/design-setup-site

Screen Shot 2021-09-28 at 11 01 41 am


No console errors or anything to report.

----

After about 90-120s it ended up loading My Home...
Maybe just taking longer to headstart because of the number of posts and including media?

**Comment by p-jackson** [comment]
> After about 90-120s it ended up loading My Home...
> Maybe just taking longer to headstart because of the number of posts and including media?

Yes sometimes the headstart process can take a super long time sometimes! So it probably is to do with how many posts this theme's demo has. Sometimes the headstart times aren't so bad though. It seems intermitent, for me at least.

**Comment by p-jackson** [comment]
> After about 90-120s it ended up loading My Home

Maybe I can try and do some profiling to figure out why it's so slow. When I run headstart via the command line it's still slow, but more like 6-7 seconds.
It could just be that my sandbox is faster than the production webservers since it has less to do.

File: client/signup/steps/design-picker/index.jsx [hunk 4]
```diff
@@ -49,24 +72,35 @@ class DesignPickerStep extends Component {
...
 		if ( this.props.showOnlyThemes ) {
-			// Only offering designs that are also available as themes. This means excluding
-			// designs where the `template` has a layout that's different from what the theme's
-			// default Headstart annotation provides.
-			designs = getAvailableDesigns().featured.filter(
-				( { features, template, theme } ) => theme === template && ! features.includes( 'anchorfm' )
-			);
+			// Should probably be pulled out into a utility to format theme response as design picker design
+			// taxonomies.theme_subject probably maps to category
+			designs = this.props.themes
```
Comment (arthur791004) [inline client/signup/steps/design-picker/index.jsx:80]: We can create a selector for this case and then we don't need to do filtering and mapping all the time. But we can do it in another PR.

File: client/signup/steps/design-picker/index.jsx [hunk 5]
```diff
@@ -49,24 +72,35 @@ class DesignPickerStep extends Component {
...
-			<DesignPicker
-				designs={ designs }
-				theme={ this.props.isReskinned ? 'light' : 'dark' }
-				locale={ this.props.locale } // props.locale obtained via `localize` HoC
-				onSelect={ this.pickDesign }
-				className={ classnames( {
-					'design-picker-step__is-large-thumbnails': this.props.largeThumbnails,
-				} ) }
-			/>
+			<div>
```
Comment (arthur791004) [inline client/signup/steps/design-picker/index.jsx]: Why do we need this?

Comment (michaeldcain) [reply client/signup/steps/design-picker/index.jsx]: Sorry, I think this might have been remnants from where I was experimenting with the category sidebar. It can be removed from this PR

**Comment by ianstewart** [comment]
> Turns out Ryu is the only recommended theme without annotations. I will explicitly exclude it for now, but will need to P2 about whether this is just an oversight.

Just an oversight. All the themes should have annos.

> Yes sometimes the headstart process can take a super long time sometimes! So it probably is to do with how many posts this theme's demo has.

Demo site content can be trimmed down when this is the case. Probably non-optimized images and too much content. It should really only have enough content to look like the home-page design.

**Review by michaeldcain — APPROVED** [review]
I don't see anything blocking - just the comments from Arthur.

File: client/signup/steps/design-picker/index.jsx (hunk 4 above)
Comment (p-jackson) [reply client/signup/steps/design-picker/index.jsx:80]: I've updated the comment to reflect what I actually think we should do, which is perhaps move the fetching code and filtering code into the `<DesignPicker>` itself. In fact as the idea of "design" becomes less important, I don't think we'll need this intermediary design format, and the `<DesignPicker>` can just start consuming the API response format directly.
